server/application/rest/work_shift.py

```diff
@@ -42,64 +36,41 @@ def work_shifts():
     """
     if request.method == "GET":
         repo = MemRepo(shifts)
-        try:
-            result = workshift_list_use_case(repo)
-            response_data = json.dumps(result, cls=WorkShiftJsonEncoder)
-            return Response(
-                response_data,
-                mimetype="application/json",
-                status=200
-            )
-        except (AuthenticationError, NotFoundError) as e:
-            return handle_error_response(e)
+        result = workshift_list_use_case(repo)
+        return Response(
+            json.dumps(result, cls=WorkShiftJsonEncoder),
+            mimetype="application/json",
+            status=200,
+        )
     elif request.method == "POST":
         user = get_user_from_token(request.headers)
         data = request.get_json()
         for shift in data:
             shift["worker"] = user
         repo = MemRepo(shifts)
-        try:
-            workshift_add_multiple_use_case(repo, data)
-            response_data = json.dumps(data, cls=WorkShiftJsonEncoder)
-            return Response(
-                response_data,
-                mimetype="application/json",
-                status=200
-            )
-        except (AuthenticationError, NotFoundError) as e:
-            return handle_error_response(e)
-
+        workshift_add_multiple_use_case(repo, data)
+        return Response(
+            json.dumps(data, cls=WorkShiftJsonEncoder),
+            mimetype="application/json",
+            status=200,
+        )
 
 def get_user_from_token(headers):
-    if "Authorization" in headers:
-        return headers["Authorization"]
-    raise AuthenticationError("Authentication failed")
-
-
+    return headers["Authorization"]
+    
 @blueprint.route("/shifts/<shift_id>", methods=["DELETE"])
 @cross_origin()
 def delete_work_shift(shift_id):
     try:
         shift_id = str(shift_id)
         user_email = get_user_from_token(request.headers)
         repo = MemRepo(shifts)
-        delete_shift_use_case(repo, shift_id, user_email)
-        return Response(
-            json.dumps({"message": "Shift deleted successfully"}),
-            mimetype="application/json",
-            status=200
-        )
-    except (AuthenticationError, NotFoundError) as e:
-        return handle_error_response(e)
 
+        response = delete_shift_use_case(repo, shift_id, user_email)
+        if isinstance(response, ResponseFailure):
+            return response.message, 400 if response.response_type != ResponseTypes.SYSTEM_ERROR else 500
+        return response.value, 200
```

Comment:
Instead of hard coding error numbers in each route, use the mapping from response type to error number (as in chapter 5)

Reply:
@Sai9797 In Chapter 5, there is a mapping from ResponseType (for each response type) to an error code, and the mapping is used when returning a response code (instead of hard coding 400, 500, 200, etc). Please review Chapter 5 code example and improve the solution.